grant-project/zcash-grant-system
# Post-authorization location forwarding.

Closes #46.

### What this does
- saves `onlyLoggedIn` `location` route info and uses it to forward the user after authorization
- flashes antd `message` to indicate authorization is required (generic verbiage)
- removes `requiresAuth` from `Template` component
- localizes all forwarding logic in `AuthRoute`
- stores `authForwardLocation` in `auth` redux store

### What this doesn't do
- deep links of the form `/profile/<id>?tab=invites` will not forward user to auth, however `/profile?tab=invites` will work
- deep links for PENDING proposals where a user must be a team-member and they are not logged in will not forward the user to auth

### What could be better
- could go less top-down with auth-required by using a component that can be placed anywhere and have more customized logic and behavior

### Testing
- logout
- click 'Start a Proposal', you should be forwarded to sign-in screen and an antd message should flash
- login, you should be forwarded to the create page
- logout, click 'Start a Proposal',  but this time 'Create a new account', you should again be forwarded to proposal creation screen after successful account creation
- logout, hard load `http://localhost:3000/profile?tab=pending`, login and make sure you are forwarded with the search query params

## Screen
![screen shot 2019-01-17 at 1 14 23 pm](https://user-images.githubusercontent.com/11430954/51342544-d9343680-1a59-11e9-8b57-411f554eda30.png)
